src/pybind/mgr/cephadm/services/smb.py

```diff
@@ -20,7 +20,22 @@ class SMBService(CephService):
 
     def config(self, spec: ServiceSpec) -> None:
         assert self.TYPE == spec.service_type
-        logger.warning('config is a no-op')
+        smb_spec = cast(SMBSpec, spec)
+        self._configure_cluster_meta(smb_spec)
+
+    def ranked(self, spec: ServiceSpec) -> bool:
+        smb_spec = cast(SMBSpec, spec)
+        return 'clustered' in smb_spec.features
+
+    def fence_old_ranks(
+        self,
+        spec: ServiceSpec,
+        rank_map: Dict[int, Dict[int, Optional[str]]],
+        num_ranks: int,
+    ) -> None:
+        logger.warning(
+            'fence_old_ranks: Unsupported %r %r', rank_map, num_ranks
```

Comment:
CTDB should handle this. However, it might be nice to still revoke old keys using this function in the future. I just don't think it's needed at the momemnt.